tests/test_dataclasses.py

```diff
@@ -540,19 +538,22 @@ def test_csv_values(self):
         trade = Trade(**self.initial_data)
         rate = round(trade.trade_rate, 4)
         assert (trade.csv_values() ==
-                (trade.creation_time, rate, trade.traded_energy, trade.seller, trade.buyer,
-                 trade.matching_requirements))
+                (trade.creation_time, rate, trade.traded_energy, trade.seller.name,
+                 trade.buyer.name, trade.matching_requirements))
 
     def test_to_json_string(self):
         trade = Trade(**self.initial_data)
         trade_dict = deepcopy(trade.__dict__)
-        trade_dict["offer_bid"] = trade_dict["offer_bid"].to_json_string()
+        trade_dict["match_details"]["offer"] = (
+            trade_dict["match_details"]["offer"].to_json_string())
```

Comment:
Can you please explain why ```trade_dict["match_details"]["offer"].to_json_string()``` is called instead of ```trade_dict["match_details"]["offer"].serializable_dict()```?

Reply:
This is an interesting question, and relates to the question about the offer_bid_id in the serializable_dict method. I had a second look at the code and it seems that the to_json_string / from_json are solely used for the serialization of the object towards the external connection, while the serializable_dict is used for serializing the object for the Kafka connection to the simulation results service. I know that this will prolong the PR a bit, however if you agree I would like to change this so that I unify the serializable_dict and to_json_string methods. That way we will hopefully have only one serialization flwo and not multiple in the same class.

Reply:
Thank you for checking this out. Yes, a unified serializatoin flow will be very helpful. Also, IMO, It can enhance the developer's experience by removing jsons inside a dictionary! They are really painful!